### PR TITLE
* Added preference to decide the max alert to show in the panel

### PR DESCRIPTION
**Pull request by YehonatanVonage**
* Added preference to decide if to show alerts if panel is in disable mode

### CODE REVIEW

File: src/status_ctrl.js [hunk 1]
```diff
@@ -310,6 +313,30 @@ export class StatusPluginCtrl extends MetricsPanelCtrl {
 		}
 	}
 
+	handleMaxAlertsToShow() {
+		if(this.panel.maxAlertNumber != null && this.panel.maxAlertNumber >= 0) {
+			let currentAlertsNumber = this.panel.maxAlertNumber;
+			if(this.disabled.length > currentAlertsNumber) {
+				this.disabled = this.disabled.slice(0,currentAlertsNumber);
+			}
+			currentAlertsNumber = Math.max(currentAlertsNumber - this.disabled.length, 0);
```
Comment (TomerAmirV) [inline src/status_ctrl.js]: This block seems to be repeating. could you maybe make a method out of it?

File: src/editor.html [hunk 2]
```diff
@@ -22,6 +24,31 @@ <h5 class="section-heading">{{ measurement.alias }}</h5>
 		  </span>
 			<input type="text" ng-model="ctrl.panel.namePrefix" class="gf-form-input max-width-12" ng-change="ctrl.onRender()">
 		</div>
+		<div class="gf-form">
+		  <span class="gf-form-label width-12">
+			Max Alerts
+			<info-popover mode="right-normal">
+			  Max alerts number to show in the panel. In case value is not configured, don't limit the alerts
```
Comment (TomerAmirV) [inline src/editor.html]: Maybe change "don't limit the alerts" to "show all"?

File: src/status_ctrl.js [hunk 3]
```diff
@@ -317,24 +318,19 @@ export class StatusPluginCtrl extends MetricsPanelCtrl {
 
 	handleMaxAlertsToShow() {
 		if(this.panel.maxAlertNumber != null && this.panel.maxAlertNumber >= 0) {
-			let currentAlertsNumber = this.panel.maxAlertNumber;
-			if(this.disabled.length > currentAlertsNumber) {
-				this.disabled = this.disabled.slice(0,currentAlertsNumber);
+			let currentMaxAllowedAlerts = this.panel.maxAlertNumber;
+			let filteredOutAlerts = 0;
+			let arrayNamesToSlice = ["disabled", "crit", "warn", "display"];
+			for(let i=0; i<arrayNamesToSlice.length; i++) {
```
Comment (TomerAmirV) [inline src/status_ctrl.js]: Why not use `arrayNamesToSlice.forEach((arrayName) => { ... });`?
should be much cleaner

File: src/editor.html [hunk 4]
```diff
@@ -7,6 +7,8 @@
 </div>
 
 <div class="editor-row">
+
+	<!--General section-->
 	<div class="section gf-form-group">
 		<h5 class="section-heading">{{ measurement.alias }}</h5>
```
Comment (TomerAmirV) [inline src/editor.html]: This line is out of place, please remove it

File: src/status_ctrl.js (hunk 3 above)
Comment (TomerAmirV) [inline src/status_ctrl.js]: Also, we could put all arrays under a single object, and loose the string array altogether, but that is optional.

Comment (YehonatanVonage) [inline src/status_ctrl.js]: fixed

File: src/editor.html (hunk 4 above)
Comment (YehonatanVonage) [inline src/editor.html]: fixed

Comment (TomerAmirV) [inline src/editor.html]: Wrong line... the `<h5>` line. 😆

**Comment by TomerAmirV** [comment]
👍

**Pull request merged**